include/render.hpp

```diff
@@ -140,18 +95,50 @@ template <int width, int height, int samples, int depth> class render_kernel {
     }
     // If not returned within max_depth return black
     return color { 0.0f, 0.0f, 0.0f };
-  }
+  };
+
+  return [=](int x_coord, int y_coord) -> void {
+    // map the 2D indices to a single linear, 1D index
+    const auto pixel_index = y_coord * width + x_coord;
+
+    // Color sampling for antialiasing
+    color final_color(0.0f, 0.0f, 0.0f);
+    for (auto i = 0; i < samples; i++) {
+      const auto u = (x_coord + random_float()) / width;
+      const auto v = (y_coord + random_float()) / height;
+      // u and v are points on the viewport
+      ray r = cam.get_ray(u, v);
+      final_color += get_color(r);
+    }
+    final_color /= static_cast<real_t>(samples);
 
-  // Accessor objects
-  sycl::accessor<color, 1, sycl::access::mode::write,
-                 sycl::access::target::global_buffer>
-      m_frame_ptr;
-  sycl::accessor<hittable_t, 1, sycl::access::mode::read,
-                 sycl::access::target::global_buffer>
-      m_hitable_ptr;
-  int num_hittables;
-  camera m_camera;
-};
+    // Write final color to the frame buffer global memory
+    frame_ptr[pixel_index] = final_color;
+  };
+}
+
+template <int width, int height, typename T>
+void executor(trisycl::handler& cgh, T render_kernel) {
+  if constexpr (buildparams::use_single_task) {
+    cgh.single_task([render_kernel]() -> void {
```

Comment:
You can in general remove the explicit lambda return type except when you cannot or you want to clarify the code.